main.js

```diff
@@ -250,6 +250,8 @@ class openknx extends utils.Adapter {
             rawVal = Buffer.from(state.val, "hex");
             isRaw = true;
             this.log.warn("Missing implementation for unhandeled DPT " + dpt + ", assuming raw values");
+        } else if (tools.isSwitchDPT(dpt)) {
```

Comment:
why is this special case added?  switch is boolean and handeled by line 254, as all other not handeled dpts

Reply:
If the user sends 1 or 0 instead of true or false this will not work with knx.js 
Some vis elements sending this kind of incorrect boolean

Reply:
ok, dpt 1 is imo the only bool datatype
implementation should be more generic, add it outside this block

if
    "type": "boolean",
then
 convert to true and false,

then you dont need isSwitchDPT
and also add a comment

appl. must be able to process it correctly in receive direction;